### PR TITLE
cpp aql current_user() v2

### PR DESCRIPTION
**Pull request by baslr**

### CODE REVIEW

**Comment by baslr** [comment]
see #3063

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -3724,6 +3713,32 @@ AqlValue Functions::IsSameCollection(
   return AqlValue(arangodb::basics::VelocyPackHelper::NullValue());
 }
 
+#include "Utils/ExecContext.h"
```
Comment (fceller) [inline arangod/Aql/Functions.cpp]: Please move the include to the top of the file

**Comment by baslr** [comment]
some internals have to change
req more infos @mchacki

**Comment by fceller** [comment]
@baslr

the test failed with:

[FAILED] testCurrentUser in 0.002 s: at assertion #1: assertEqual: ([ null ]) is not equal to ([ "root" ])

**Comment by mchacki** [comment]
Yes this is known issue with this branch. The reason is that without authentication turned on the JSLayer und C++ layer behave differently, namely:
JS looks into auth-header (which we send in this test) and despite of authentication returns the user in this header.
C++ takes the actual auth object (which is null without authentication) and does not have any reference to the request header any more.
Needs some internal rearrangement to be fixed.

**Comment by jsteemann** [comment]
Can you pull latest devel into this branch and try to fix it?
You can probably use the ExecContext everywhere now.

**Comment by baslr** [comment]
@jsteemann also to devel xor exclusive to devel?

**Comment by jsteemann** [comment]
devel only

**Comment by baslr** [comment]
new PR https://github.com/arangodb/arangodb/pull/3498

**Pull request closed without merging**